datadog-monitors-downtime-handler/src/datadog_monitors_downtime/handlers.py

```diff
@@ -0,0 +1,199 @@
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL,
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = downtimes_api.DowntimesApi(api_client)
+        try:
+            api_resp = api_instance.create_downtime(downtime_body)
+            model.Id = api_resp.id
+        except ApiException as e:
+            LOG.error("Exception when calling DowntimeApi->create_downtime: %s\n" % e)
+            return ProgressEvent(status=OperationStatus.FAILED, resourceModel=model, message=e.body)
+
+    return read_handler(session, request, callback_context)
+
+
+@resource.handler(Action.UPDATE)
+def update_handler(
+    session: Optional[SessionProxy],
+    request: ResourceHandlerRequest,
+    callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS,
+        resourceModel=model,
+    )
+
+    LOG.info(f"Starting the {TYPE_NAME} Update Handler")
+
+    downtime_body = build_downtime_struct(model)
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL,
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = downtimes_api.DowntimesApi(api_client)
+        try:
+            api_instance.update_downtime(model.Id, downtime_body)
+        except ApiException as e:
+            LOG.error("Exception when calling DowntimeApi->update_downtime: %s\n" % e)
+            return ProgressEvent(status=OperationStatus.FAILED, resourceModel=model, message=e.body)
+
+    return read_handler(session, request, callback_context)
+
+
+@resource.handler(Action.DELETE)
+def delete_handler(
+    session: Optional[SessionProxy],
+    request: ResourceHandlerRequest,
+    callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS,
+        resourceModel=model,
+    )
+    LOG.info(f"Starting the {TYPE_NAME} Delete Handler")
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL,
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = downtimes_api.DowntimesApi(api_client)
+        try:
+            api_instance.cancel_downtime(model.Id)
+        except ApiException as e:
+            LOG.error("Exception when calling DowntimeApi->cancel_downtime: %s\n" % e)
+            return ProgressEvent(status=OperationStatus.FAILED, resourceModel=model, message=e.body)
+
+    return read_handler(session, request, callback_context)
```

Comment:
Should we call the read handler on delete ?

Reply:
Good catch!